website/static/js/nodesDelete.js

```diff
@@ -280,24 +282,105 @@ NodesDeleteViewModel.prototype.back = function() {
     this.page(this.SELECT);
 };
 
-function NodesDelete(selector, node) {
+/**
+ * view model which corresponds to nodes_delete.mako (#nodesDelete)
+ * Used for quickly deleting nodes with no children.
+ * @type {QuickDeleteViewModel}
+ */
+
+var QuickDeleteViewModel = function(nodeType, isPreprint, nodeApiUrl){
     var self = this;
+    self.confirmationString = $osf.getConfirmationString();
+    self.QUICKDELETE = 'quickDelete';
+    self.SELECT = 'select';
+    self.CONFIRM = 'confirm';
+    self.page = ko.observable(self.QUICKDELETE);
+    self.confirmInput = ko.observable('');
+    self.canDelete = ko.observable(false);
+    self.nodeType = nodeType;
+    self.nodeApiUrl = nodeApiUrl;
+    self.isPreprint = isPreprint;
+    self.message = ko.computed(function(){
+        var preprintMessage = '<br><br>This ' + self.nodeType + ' contains a <strong>preprint</strong>. Deleting this ' +
+          self.nodeType + ' will also delete your <strong>preprint</strong>. This action is irreversible.';
+        var message = 'It will no longer be available to other contributors on the project.' +
+        (self.isPreprint ? preprintMessage : '');
+
+        return message;
+    });
+    self.atMaxLength = ko.observable(false);
+    self.nodesDeleted = ko.observable(true);
+    self.pageTitle = ko.computed(function() {
+        return 'Are you sure you want to delete this ' + self.nodeType + '?';
+    });
+};
 
-    self.selector = selector;
-    self.$element = $(self.selector);
-    self.viewModel = new NodesDeleteViewModel(node);
-    self.viewModel.fetchNodeTree().done(function(response) {
-        new NodesDeleteTreebeard('nodesDeleteTreebeard', response, self.viewModel.nodesState, self.viewModel.nodesOriginal);
+QuickDeleteViewModel.prototype.clear = function(){
+    var self = this;
+    self.nodesDeleted = ko.observable(false);
+};
+
+QuickDeleteViewModel.prototype.handleEditableUpdate = function(element) {
```

Comment:
Is there any way to dry up some of this duplication? The QuickDeleteViewModel and NodesDeleteModel have a lot of identical sections. This, for example, is identical to `NodesDeleteViewModel.prototype.handleEditableUpdate`